org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileManager.java

```diff
@@ -60,7 +59,10 @@ public void loadFiles() throws UserFacingException, UserInitiatedException {
    */
   @Override
   public ProjectFile getFile() {
-    // TODO: Enable multi-file. For now, we will always have exactly one file.
+    // TODO: Enable multi-file. For now, load the first file.
+    if (fileList == null) {
+      return null;
```

Comment:
Trying to think through the scenario where this would happen. Should this throw instead?

Reply:
Good question, I also wasn't sure about this. The only reasons it would occur is if the Javabuilder code was calling the functions incorrectly (calling get before load), or the load failed. If the load failed we would throw an exception earlier to the end user. It felt like a lot of exception declaration to add a throw here, but I can switch it if that is our preferred pattern. Thinking about it more this doesn't cover the case where the 'code' loaded successfully but it was an empty object--then we will get an index out of bound exception.

Reply:
this function will be obsolete once we enable multi-file, since we will switch to returning the entire list, so I would lean towards not adding complexity here.

Reply:
We'd want to keep what's happening clear to the user and the developer. Exceptions have the benefit of being pretty verbose and customizable as to what's happening. That doesn't mean we need to use them in every case.

Reply:
Sorry, my browser must have had this thread cached, and I didn't see your most recent comment. 
✔️  Sounds good to me!